### PR TITLE
DeepSeek MLA: absorb low-rank up-projection into Q for eager/SDPA (V2/V3/V3.2)

### PR DESCRIPTION
**Pull request by zaid646**
[CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=47011)

## Summary

The standard MLA forward materializes the full decompressed K and V tensors (e.g., `kv_lora_rank=512` -> `num_heads * (qk_nope_head_dim + v_head_dim)` = up to 32768) before calling the attention interface. This is wasteful for eager/SDPA: the `kv_b_proj` up-projection can be absorbed into the Q-nope and output projections, working entirely in the compressed `kv_lora_rank` space.

This PR applies the **absorbed attention** optimization to all three MLA model families:
- **DeepSeek-V2** (`modular_deepseek_v2.py`)
- **DeepSeek-V3** (`modular_deepseek_v3.py`)
- **DeepSeek-V3.2 (DSA)** (`modular_deepseek_v32.py`)

### How it works

Split `kv_b_proj.weight` into two sub-matrices:
- `W_uk` maps compressed k to the nope key-subspace
- `W_uv` maps compressed k to the value subspace

**Forward:**
absorbed_q = q_nope @ W_uk^T
scores = absorbed_q @ compressed_k^T + RoPE scores
output = softmax(scores) @ compressed_k @ W_uv^T

Full K/V is **only** materialized when a KV-cache update is required. Flash-attention and decode steps with pre-existing caches keep the original expanded path.

### Testing

- Unit tests verify the absorbed path matches the standard expansion to 1e-5 for V2, V3, and V3.2
- Full model forward (no MoE, eager mode) produces valid logits with no NaN
- Cache path tested: prefill (absorbed) + decode (standard) round-trip matches no-cache baseline exactly
- Flash-attention and decode-with-cache paths are unchanged (fall through to the original expansion)

### CODE REVIEW

**Comment by zaid646** [comment]
@huggingface/transformers-maintainers both `Check repository consistency` and `Check code quality` keep failing with a self-hosted runner infrastructure error:

> Executing the custom container implementation failed. Please contact your self hosted runner administrator.

This isn't a code issue — the runner itself can't execute custom containers. Could someone re-trigger CI or check the runner setup? First commit was bd242ac, empty commit 42507cd also failed the same way. Thanks!
